nilearn/plotting/image/_utils.py

```diff
@@ -0,0 +1,164 @@
+    # Having a header is required by the load_niimg function
+    header = None  # type: ignore[assignment]
+
+    def __init__(self, data=None, affine=None, header=None):
+        # Comply with spatial image requirements while allowing empty init
+        pass
+
+    def load(self):
+        if self.data is None:
+            anat_img = load_mni152_template(resolution=2)
+            anat_img = reorder_img(anat_img, copy_header=True)
+            data = get_data(anat_img)
+            data = data.astype(np.float64)
+            anat_mask = binary_fill_holes(data > np.finfo(float).eps)
+            data = np.ma.masked_array(data, np.logical_not(anat_mask))
+            self._affine = anat_img.affine
+            self.data = data
+            self.vmax = data.max()
+            self._shape = anat_img.shape
+
+    @property
+    def _data_cache(self):
+        self.load()
+        return self.data
+
+    @property
+    def _dataobj(self):
+        self.load()
+        return self.data
+
+    def get_data(self):
+        self.load()
+        return self.data
+
+    @property
+    def affine(self):
+        self.load()
+        return self._affine
+
+    def get_affine(self):
+        self.load()
+        return self._affine
+
+    @property
+    def shape(self):
+        self.load()
+        return self._shape
+
+    def get_shape(self):
+        self.load()
+        return self._shape
+
+    def __str__(self):
+        return "<MNI152Template>"
+
+    def __repr__(self):
+        return "<MNI152Template>"
+
+
+# The constant that we use as a default in functions
+MNI152TEMPLATE = _MNI152Template()
+
+
+def get_cropped_cbar_ticks(cbar_vmin, cbar_vmax, threshold=None, n_ticks=5):
+    """Return ticks for cropped colorbars."""
+    new_tick_locs = np.linspace(cbar_vmin, cbar_vmax, n_ticks)
+    if threshold is not None:
+        # Case where cbar is either all positive or all negative
+        if 0 <= cbar_vmin <= cbar_vmax or cbar_vmin <= cbar_vmax <= 0:
+            idx_closest = np.argmin(
+                [abs(abs(new_tick_locs) - threshold) for _ in new_tick_locs]
+            )
+            new_tick_locs[idx_closest] = threshold
+        # Case where we do a symmetric thresholding
+        # within an asymmetric cbar
+        # and both threshold values are within bounds
+        elif cbar_vmin <= -threshold <= threshold <= cbar_vmax:
+            new_tick_locs = get_cbar_ticks(
+                cbar_vmin, cbar_vmax, threshold, n_ticks=len(new_tick_locs)
+            )
+        # Case where one of the threshold values is out of bounds
+        else:
+            idx_closest = np.argmin(
+                [abs(new_tick_locs - threshold) for _ in new_tick_locs]
+            )
+            new_tick_locs[idx_closest] = (
+                -threshold if threshold > cbar_vmax else threshold
+            )
+    return new_tick_locs
+
+
+def load_anat(anat_img=MNI152TEMPLATE, dim="auto", black_bg="auto"):
+    """Load anatomy, for optional diming."""
```

Comment:
I've put `load_anat` in `_utils` but I export in `__init__.py` as it is used in other packages. So it can be used by importing from `nilearn.plotting.image`. Is there an incovenience?

Reply:
not an inconvenience but a change from what is done in the rest of the code base where we don't expose private functions in a subpackage

Reply:
so maybe better not put it in the init

Reply:
Then both for `MNI152TEMPLATE` and `load_anat` we have the same problem. If I change `image._utils` to `image.utils`, that would solve the problem for both. How does it sound to you?